### DISCUSSION TITLE
RANS k-ε model

### DISCUSSION THREAD

**Discussion by AdelineHunter** [idaholab/moose#21480] [Q&A Modules: Navier-Stokes] [answered]
Hi Everyone,
![image](https://user-images.githubusercontent.com/108536058/176915190-1bb8b116-abd4-4522-86f5-bc0dde9c3194.png)
First of all, thank you to all the workers in Idaho. MOOSE has really helped me a lot.
And I learned from Workshop Slides that a coarse regularized k-epsilon model will be added soon. What I want to know is whether the model is based on FEM? And how soon can it be added?
Besides, I feel that no new model has been added to the FEM Navier-Stokes module for a long time. I don't know whether the module will be updated in the future.
Thank you for your time.

**Comment by GiudGiud** [comment, marked as answer]
Hello

The model will be based on finite volume. 
It ll be added this summer. 
At INL we are no longer expanding the finite element Navier Stokes capability. For our applications it has proved unsatisfactory and the finite volume work is our focus right now. We are still accepting and would gladly review contributions from the community on the finite element side. 


Guillaume

**Reply by AdelineHunter** [reply]
Hello,
`It has proved unsatisfactory.`
Does this mean that the finite element research in CFD is not in-depth enough, resulting in the poor robustness of this method? Or because of the program structure, it may be unexpectedly difficult for finite element CFD to continue to develop based on MOOSE.

If it is the first reason, I think I will continue to do this and provide open source code to the community in the future.
If it's the second reason, it's really a shame, because I think MOOSE is the most beautiful of the finite element frameworks so far

Thank you for your time.

**Reply by GiudGiud** [reply]
MOOSE didnt preclude implementing advanced schemes for finite element CFD. The unsatisfactory parts were the lack of local conservation in our CFEM implementation (DG schemes can be conservative), the sharps peaks near flow corners, and the ambiguity at corners of fixed velocity inlets with no-slip walls (what value do you pick for the corner node?)

I dont mean to discourage your research or adding much-needed turbulence models to our FEM CFD.

Some of the developments we are working on, such as segregated solver capabilities in MOOSE, will also be applicable to finite element, benefitting both implementation's performance

**Reply by AdelineHunter** [reply]
Hello, Guillaume
Segregated solver capabilities in MOOSE. That is an exciting capability. How soon will this capability be implemented in MOOSE?

**Reply by GiudGiud** [reply]
within a year most likely.

**Reply by AdelineHunter** [reply]
Thank you, Mr. Guillaume

Look forward to more features, thanks to the Idaho staff.

By the way. Where can I get some of the new features for Navier-Stokes? Although the code may not have passed the rigorous testing in Idaho, I want to get a taste of some of these features.
Or some other finite element program for Navier-Stokes that was previously developed but not posted to Github, even though it's not completely finished.

Thanks.
